test/utils.sh

```diff
@@ -227,3 +227,134 @@ get_image_manifests() {
   echo "${image_manifests}"
 
 }
+
+# This function will be used by tekton tasks in build-definitions
+# It returns the ocp_version targeted by the FBC fragment
+get_ocp_version_from_fbc_fragment() {
+  local FBC_FRAGMENT="$1"
+
+  if [ -z "$FBC_FRAGMENT" ]; then
+    echo "Missing FBC_FRAGMENT parameter" >&2
+    exit 2
+  fi
+
+  #Get target ocp version from the fragment
+  local ocp_version
+  if ! ocp_version=$(skopeo inspect --no-tags --raw docker://"$FBC_FRAGMENT"); then
+    echo "Could not inspect image $FBC_FRAGMENT"
+    exit 1
+  fi
+
+  ocp_version=$(echo "$ocp_version" |  jq -r '.annotations."org.opencontainers.image.base.name"' | cut -d: -f2)
+  echo "$ocp_version"
```

Comment:
In the FBC validation, we have a different way to get the version: https://github.com/konflux-ci/build-definitions/blob/main/task/fbc-validation/0.1/fbc-validation.yaml#L214-L217

Reply:
hah, here it's used the way I have it in my code : https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/prepare-fbc-release/prepare-fbc-release.yaml#L92-L99

which way would you like me to go?

Reply:
According to the ADR, yours should be sufficient. The method in the validation task is more robust as it handles `repository:tag@sha256:digest`, but I don't know if it is even possible for buildah to produce the `org.opencontainers.image.base.name` annotation with that format.

Reply:
According to the ADR, there are two separate annotations for the `digest` and `name`

```
  "annotations": {
    "org.opencontainers.image.base.digest": "sha256:e5a07eff6865b2761889ee275d9fc940237c90d05d63b00f60350841ecf42df2",
    "org.opencontainers.image.base.name": "registry.redhat.io/openshift4/ose-operator-registry:v4.12"
  }
  ```

If you don't feel strongly against it, I would like to keep ocp_version extraction as it is until explicitly changed in the ADR.

Reply:
That is fine. We will need to ensure that this method is tested before changing the current verification logic over to using the function.

Reply:
makes sense. thanks!

Reply:
@arewm I did change my code to handle the `repository:tag@sha256:digest` case. Please let me know if this is good.